name-service/js/package.json

```diff
@@ -23,26 +23,36 @@
     "prepublish": "tsc",
     "lint": "yarn pretty && eslint --max-warnings 0 'src/*.ts'",
     "lint:fix": "yarn pretty:fix && eslint 'src/*.ts' --fix",
-    "pretty": "prettier --check 'src/*.ts'",
-    "pretty:fix": "prettier --write 'src/*.ts'",
-    "doc": "yarn typedoc src/index.ts"
+    "pretty": "prettier --check '{src/*.ts,test/*/*.ts}'",
+    "pretty:fix": "prettier --write '{src/*.ts,test/*/*.ts}'",
+    "doc": "yarn typedoc src/index.ts",
+    "test": "yarn test:unit && yarn test:e2e",
+    "test:unit": "mocha test/unit",
+    "test:e2e": "start-server-and-test 'solana-test-validator --bpf-program namesLPneVptA9Z5rqUDD9tMTWEJwofgaYwp8cawRkX ../../target/deploy/spl_name_service.so --reset --quiet' http://localhost:8899/health 'mocha test/e2e'"
   },
   "prettier": {
     "singleQuote": true
   },
   "devDependencies": {
     "@tsconfig/recommended": "^1.0.1",
+    "@types/chai": "^4.3.4",
+    "@types/chai-as-promised": "^7.1.5",
+    "@types/mocha": "^10.0.1",
     "@types/node": "^14.14.20",
     "@typescript-eslint/eslint-plugin": "^4.0.1",
     "@typescript-eslint/parser": "^4.0.1",
     "babel-eslint": "^10.1.0",
+    "chai": "^4.3.7",
+    "chai-as-promised": "^7.1.1",
     "eslint": "^7.8.0",
     "eslint-config-prettier": "^6.11.0",
     "eslint-plugin-eslint-comments": "^3.2.0",
     "eslint-plugin-functional": "^3.0.2",
     "eslint-plugin-import": "^2.22.0",
+    "mocha": "^10.2.0",
     "prettier": "^2.2.1",
-    "ts-node": "^9.1.1",
+    "start-server-and-test": "^1.15.3",
+    "ts-node": "^10.9.1",
```

Comment:
have to update ts-node since it was giving error when running mocha with yarn. don't particularly know why but error was ``` Debug Failure. False expression: Non-string value passed to `ts.resolveTypeReferenceDirective` ```. Found on stackoverflow it might be due to ts-node version mismatch.